src/reducers/versions.spec.tsx

```diff
@@ -117,12 +119,84 @@ describe(__filename, () => {
 
       expect(state).toEqual({
         ...initialState,
+        expandedPaths: getParentFolders(version.file.selected_file),
         versionInfo: {
           [version.id]: createInternalVersion(version),
         },
       });
     });
 
+    it('prunes expandedPaths when load version info', () => {
+      const path1 = 'path/1';
+      const path2 = 'path/2';
+      const expandedPaths = [path1, path2, ROOT_PATH];
+      const oldVersionId = nextUniqueId();
+      const oldVersion = createExternalVersionWithEntries(
+        [
+          createFakeEntry('directory', path1),
+          createFakeEntry('directory', path2),
+        ],
+        { id: oldVersionId },
+      );
+      const newVersion = createExternalVersionWithEntries(
+        [createFakeEntry('directory', path1)],
+        { id: nextUniqueId() },
+      );
+
+      let state = reducer(
+        undefined,
+        actions.loadVersionInfo({ version: oldVersion }),
+      );
+      state = reducer(state, actions.expandTree({ versionId: oldVersionId }));
+      state = reducer(state, actions.loadVersionInfo({ version: newVersion }));
+
+      expect(state).toHaveProperty(
+        'expandedPaths',
+        getPrunedExpandedPaths(
+          expandedPaths,
+          createInternalVersion(newVersion),
+        ),
+      );
+    });
+
+    it('prunes expanded paths when setCurrentVersionId is dispatched', () => {
```

Comment:
This should use dependency injection as described above, to ensure that responsibility is passed on to `getPrunedExpandedPaths` correctly.